src/careamics/lightning/callbacks/prediction_writer_callback/prediction_writer_callback.py

```diff
@@ -0,0 +1,231 @@
+        initialisation of `PredictionWriterCallback`.
+
+        Parameters
+        ----------
+        write_type : {"tiff", "custom"}
+            The data type to save as, includes custom.
+        tiled : bool
+            Whether the prediction will be tiled or not.
+        write_func : WriteFunc, optional
+            If a known `write_type` is selected this argument is ignored. For a custom
+            `write_type` a function to save the data must be passed. See notes below.
+        write_extension : str, optional
+            If a known `write_type` is selected this argument is ignored. For a custom
+            `write_type` an extension to save the data with must be passed.
+        write_func_kwargs : dict of {{str: any}}, optional
+            Additional keyword arguments to be passed to the save function.
+        dirpath : pathlib.Path or str, default="predictions"
+            The path to the directory where prediction outputs will be saved. If
+            `dirpath is not absolute it is assumed to  be relative to current working
+            directory. Nested directories will not be automatically created.
+
+        Returns
+        -------
+        PredictionWriterCallback
+            Callback for writing predictions.
+        """
+        write_strategy = create_write_strategy(
+            write_type=write_type,
+            tiled=tiled,
+            write_func=write_func,
+            write_extension=write_extension,
+            write_func_kwargs=write_func_kwargs,
+        )
+        return cls(write_strategy=write_strategy, dirpath=dirpath)
+
+    def _init_dirpath(self, dirpath):
+        """
+        Initialize directory path. Should only be called from `__init__`.
+
+        Parameters
+        ----------
+        dirpath : pathlib.Path
+            See `__init__` description.
+        """
+        dirpath = Path(dirpath)
+        if not dirpath.is_absolute():
+            dirpath = Path.cwd() / dirpath
+            logger.warning(
+                "Prediction output directory is not absolute, absolute path assumed to"
+                f"be '{dirpath}'"
+            )
+        self.dirpath = dirpath
+
+    def setup(self, trainer: Trainer, pl_module: LightningModule, stage: str) -> None:
+        """
+        Create the prediction output directory when predict begins.
+
+        Called when fit, validate, test, predict, or tune begins.
+
+        Parameters
+        ----------
+        trainer : Trainer
+            PyTorch Lightning trainer.
+        pl_module : LightningModule
+            PyTorch Lightning module.
+        stage : str
+            Stage of training e.g. 'predict', 'fit', 'validate'.
+        """
+        super().setup(trainer, pl_module, stage)
+        if stage == "predict":
+            # make prediction output directory
+            if not self.dirpath.is_dir():
+                logger.info("Making prediction output directory.")
+                self.dirpath.mkdir()
```

Comment:
Here, if it exists, it will raise an error. 

We should probably use `self.dirpath.mdkir(parents=True, exist_ok=True)`.

Reply:
There is is the `if not self.dirpath.is_dir()` condition above, but this is more elegant so I will update.